pkg/gatherers/workloads/gather_workloads_runtime_infos.go

```diff
@@ -0,0 +1,185 @@
+	coreClient corev1client.CoreV1Interface,
+) ([]podWithNodeName, error) {
+	pods, err := coreClient.Pods(os.Getenv("POD_NAMESPACE")).
+		List(ctx, metav1.ListOptions{
+			LabelSelector: "app.kubernetes.io/name=insights-runtime-extractor",
+		})
+	if err != nil {
+		return nil, err
+	}
+
+	var runtimePods []podWithNodeName
+	for _, pod := range pods.Items {
+		runtimePods = append(runtimePods, podWithNodeName{
+			podIP:    pod.Status.PodIP,
+			nodeName: pod.Spec.NodeName,
+		})
+	}
+	return runtimePods, nil
+}
+
+// Merge the workloads from a single node into the global map
+func mergeWorkloads(global workloadRuntimes,
+	node workloadRuntimes,
+) {
+	for cInfo, cRuntimeInfo := range node {
+		global[cInfo] = cRuntimeInfo
+	}
+}
+
+// Transform the workload data from the insights operator runtime component (represented by
+// a map of namespace/pod-name/container-id/insightsWorkloadRuntimeInfo) into the data
+// stored by the Insights operator as a map of namespace/pod-name/container-id/workloadRuntimeInfoContainer
+// where each value is hashed.
+func transformWorkload(h hash.Hash,
+	node nodeRuntimeInfo,
+) workloadRuntimes {
+
+	result := make(workloadRuntimes)
+
+	for nsName, nsRuntimeInfo := range node {
+		cInfo := containerInfo{
+			namespace: nsName,
+		}
+		for podName, podRuntimeInfo := range nsRuntimeInfo {
+			cInfo.pod = podName
+			for containerId, containerRuntimeInfo := range podRuntimeInfo {
+				cInfo.containerID = containerId
+				hashedContainerInfo := workloadRuntimeInfoContainer{
+					Os:              hashString(h, containerRuntimeInfo.OSReleaseID),
+					OsVersion:       hashString(h, containerRuntimeInfo.OSReleaseVersionID),
+					Kind:            hashString(h, containerRuntimeInfo.RuntimeKind),
+					KindVersion:     hashString(h, containerRuntimeInfo.RuntimeKindVersion),
+					KindImplementer: hashString(h, containerRuntimeInfo.RuntimeKindImplementer),
+				}
+
+				runtimeInfos := make([]RuntimeComponent, len(containerRuntimeInfo.Runtimes))
+				for i, runtime := range containerRuntimeInfo.Runtimes {
+					runtimeInfos[i] = RuntimeComponent{
+						Name:    hashString(h, runtime.Name),
+						Version: hashString(h, runtime.Version),
+					}
+				}
+				hashedContainerInfo.Runtimes = runtimeInfos
+				result[cInfo] = hashedContainerInfo
+			}
+		}
+	}
+	return result
+}
+
+// hashString return a hash of the string if it is not empty (or the empty string otherwise)
+func hashString(h hash.Hash, s string) string {
+	if s == "" {
+		return s
+	}
+	return workloadHashString(h, s)
+}
+
+// Get all WorkloadRuntimeInfos for a single Node (using the insights-runtime-extractor pod running on this node)
+// FIXME return an (workloadRuntimes, error)
+func getNodeWorkloadRuntimeInfos(
+	ctx context.Context,
+	h hash.Hash,
+	runtimePodIP string,
+) workloadRuntimes {
+
+	extractorURL := fmt.Sprintf("http://%s:8000/gather_runtime_info", runtimePodIP)
+	request, err := http.NewRequestWithContext(ctx, "GET", extractorURL, nil)
```

Comment:
I think it would be good to narrow down the context here used for the HTTP request. I mean something like:
```
httpCtx := context.WithTimeout(ctx, 2*time.Minute)
```
because the input `ctx` parameter has timeout in hours.

Reply:
fixed, thanks